client/state/selectors/is-requesting-media.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import MediaQueryManager from 'lib/query-manager/media';
+
+/**
+ * Returns true if media is being requested for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {bool}           True if media is being requested
+ */
+export default function isRequestingMedia( state, siteId, query ) {
+	const queryRequests = state.media.queryRequests[ siteId ];
+	const stringifiedQuery = MediaQueryManager.QueryKey.stringify( query );
+
+	if ( ! queryRequests ) {
+		return false;
+	}
+
+	return queryRequests.indexOf( stringifiedQuery ) !== -1;
```

Comment:
Personal preference, but I find [`_.includes`](https://lodash.com/docs/4.17.2#includes) to be a more human-readable equivalent, which also happens to automatically account for a falsey `queryRequests`.